### PR TITLE
Integration settings for twilio

### PR DESCRIPTION
**Pull request by SanderGi**
Task: https://app.asana.com/0/1203067047205953/1207805759212356
No connect UI, just the integration settings as requested.

### Q/A checklist

- [x] If you add new dependencies, did you update the lock file?
```bash
poetry lock --no-update
```
- [x] Run tests 
```bash
ulimit -n unlimited && ./scripts/run-tests.sh
```
- [x] Do a self code review of the changes - Read the diff at least twice.  
- [x] Carefully think about the stuff that might break because of this change - this sounds obvious but it's easy to forget to do "Go to references" on each function you're changing and see if it's used in a way you didn't expect. 
- [x] The relevant pages still run when you press submit
- [x] The API for those pages still work (API tab)
- [x] The public API interface doesn't change if you didn't want it to (check API tab > docs page)
- [x] Do your UI changes (if applicable) look acceptable on mobile?
- [x] Ensure you have not regressed the import time unless you have a good reason to do so. 
You can visualize this using tuna:
```bash
python3 -X importtime -c 'import server' 2> out.log && tuna out.log
```
To measure import time for a specific library:
```bash
$ time python -c 'import pandas'

________________________________________________________
Executed in    1.15 secs    fish           external
   usr time    2.22 secs   86.00 micros    2.22 secs
   sys time    0.72 secs  613.00 micros    0.72 secs
```
To reduce import times, import libraries that take a long time inside the functions that use them instead of at the top of the file:
```python
def my_function():
    import pandas as pd
    ...
```

### Legal Boilerplate

Look, I get it. The entity doing business as “Gooey.AI” and/or “Dara.network” was incorporated in the State of Delaware in 2020 as Dara Network Inc. and is gonna need some rights from me in order to utilize my contributions in this PR. So here's the deal: I retain all rights, title and interest in and to my contributions, and by keeping this boilerplate intact I confirm that Dara Network Inc can use, modify, copy, and redistribute my contributions, under its choice of terms.

### CODE REVIEW

**Comment by devxpy** [comment]
Just a thought - can we remove twilio tts setting from bot integration, given that we already list the google voices in the saved run? If we already support google tts, we can just change the logic to say if the saved run has selected google tts, we use twilio inbuilt google tts with the same voice. 
we can add support for polly too in time but not needed for now.

similarly for the asr model, if its set to auto detect in the saved run, we can pick up the language setting and use that for twilio too


I have seen in user testing that its hard to have 2 places to configure things + this makes it slightly easier to test without releasing the bot integration

this looks pretty good to me otherwise

**Comment by SanderGi** [comment]
> Just a thought - can we remove twilio tts setting from bot integration, given that we already list the google voices in the saved run? If we already support google tts, we can just change the logic to say if the saved run has selected google tts, we use twilio inbuilt google tts with the same voice. we can add support for polly too in time but not needed for now.
> 
> similarly for the asr model, if its set to auto detect in the saved run, we can pick up the language setting and use that for twilio too
> 
> I have seen in user testing that its hard to have 2 places to configure things + this makes it slightly easier to test without releasing the bot integration
> 
> this looks pretty good to me otherwise

What about the current use case for these settings where asr/tts isn’t enabled on the underlying gooey run? Should we just default to an arbitrary twilio tts voice and english for asr?

**Comment by devxpy** [comment]
> > Just a thought - can we remove twilio tts setting from bot integration, given that we already list the google voices in the saved run? If we already support google tts, we can just change the logic to say if the saved run has selected google tts, we use twilio inbuilt google tts with the same voice. we can add support for polly too in time but not needed for now.
> > similarly for the asr model, if its set to auto detect in the saved run, we can pick up the language setting and use that for twilio too
> > I have seen in user testing that its hard to have 2 places to configure things + this makes it slightly easier to test without releasing the bot integration
> > this looks pretty good to me otherwise
> 
> What about the current use case for these settings where asr/tts isn’t enabled on the underlying gooey run? Should we just default to an arbitrary twilio tts voice and english for asr?

Yup, the default we hardcoded now can work. We can also force people to do this at time of creation of the twilio bot, but probably ok to use the hardcoded defaults right now

**Comment by devxpy** [comment]
I just realised the user language support list can be different from the Twilio asr supported languages. 

If such an edge case happens, just fallback to gooey asr

File: daras_ai_v2/asr.py [hunk 1]
```diff
@@ -440,6 +429,30 @@ def asr_language_selector(
     )
 
 
+def get_language_match(lang: str | None, languages: list[str]) -> str | None:
```
Comment (devxpy) [inline daras_ai_v2/asr.py:432]: why not use `normalised_lang_in_collection` ??

File: routers/twilio_api.py [hunk 2]
```diff
@@ -41,6 +43,42 @@
...
 
+def get_twilio_tts_voice(bi: BotIntegration) -> str:
+    from recipes.TextToSpeech import TextToSpeechProviders
+
+    run = bi.get_active_saved_run()
+    state: dict = run.state
+    tts_provider = state.get("tts_provider")
+    if tts_provider and tts_provider == TextToSpeechProviders.GOOGLE_TTS.name:
+        voice = "Google." + state.get("google_voice_name", "en-US-Wavenet-F")
+        if voice not in TWILIO_SUPPORTED_VOICES:
```
Comment (devxpy) [inline routers/twilio_api.py:54]: why is use_gooey_tts True for this case?

File: daras_ai_v2/asr.py (hunk 1 above)
Comment (SanderGi) [reply daras_ai_v2/asr.py:432]: My bad. Didn’t realize we had a function for this. I looked at the asr code and only saw the place were we manually did it inline without calling a function so thought we didn’t have this refactored yet

File: routers/twilio_api.py (hunk 2 above)
Comment (SanderGi) [reply routers/twilio_api.py:54]: Is it? I have the following python in there that shouldn’t set it to true? It seems to work for me, I’ll double check

self.use_gooey_tts = (
            tts_provider and tts_provider != TextToSpeechProviders.GOOGLE_TTS.name
        )

File: daras_ai_v2/asr.py (hunk 1 above)
Comment (SanderGi) [reply daras_ai_v2/asr.py]: [01e3c81](https://github.com/GooeyAI/gooey-server/pull/414/commits/01e3c81a1c3afe488e24192ca928ddc5728a289b)

File: routers/twilio_api.py (hunk 2 above)
Comment (SanderGi) [reply routers/twilio_api.py:54]: [e567511](https://github.com/GooeyAI/gooey-server/pull/414/commits/e56751148d698fcfc89d117e23fc1bf1ebe4b113)

**Comment by devxpy** [comment]
image
??

**Comment by SanderGi** [comment]
> image ??

Is this before or after your refactor? You probably just set the phone number sid incorrectly in the db

**Pull request merged**